plugins/editor/key-extension.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const keyExtension = {
+  setup: function(app, cm) {
+
+    const toRegister = {
+      findNext: function() {
+        cm.execCommand('findNext');
+      },
+      findPrevious: function() {
+        cm.execCommand('findPrev');
+      },
+      moveByScrollUpLine: function() {
+        const scrollbox = cm.getScrollInfo();
+        cm.scrollTo(null, scrollbox.top - cm.defaultTextHeight());
+      },
+      moveByScrollDownLine: function() {
+        const scrollbox = cm.getScrollInfo();
+        cm.scrollTo(null, scrollbox.top + cm.defaultTextHeight());
+      },
+      Tab: function() {
```

Comment:
these are being used to tap into CodeMirror command functionality. 

Ctrl + up should use CodeMirror's command API to get the current scroll location and move up by whatever the CodeMirror default text height is.

after searching with Ctrl + f (CodeMirror built in search function), this maps F3 and Shit + F3 to find next or find previous.